packages/mobile/src/transactions/actions.ts

```diff
@@ -138,7 +140,9 @@ export const navigateToPaymentTransferReview = (
       headerText = i18n.t('walletFlow5:transactionHeaderEscrowSent')
       break
     case TokenTransactionType.Received:
-      headerText = i18n.t('walletFlow5:transactionHeaderReceived')
+      headerText = addressToDisplayName[confirmationProps.address || '']?.isCeloRewardSender
+        ? i18n.t('walletFlow5:transactionHeaderCeloReward')
+        : i18n.t('walletFlow5:transactionHeaderReceived')
```

Comment:
Nothing to do now, but I now realise this header text logic being in the action is what is causing these hoops with passing more of the redux state. It should probably be moved out of this and dealt in the screen/component displaying this.

Reply:
Yeah, it does look a bit strange having it like this. I opened an issue to track this here: #6406